pallets/domains/src/lib.rs

```diff
@@ -419,58 +370,87 @@ pub mod pallet {
             let meta = Self::require_domain(domain_lc.clone())?;
+            let owner = meta.owner.clone();
 
             if let Some(should_be_owner) = check_ownership {
                 Self::ensure_allowed_to_update_domain(&meta, &should_be_owner)?;
             }
 
-            ensure!(meta.inner_value != new_value_opt, Error::<T>::InnerValueNotChanged);
-
-            if let Some(old_value) = &meta.inner_value {
-                DomainByInnerValue::<T>::remove(&meta.owner, old_value);
-            }
+            let (old_depositor, old_deposit) = if let Some(value_with_deposit_info) =
+                DomainRecords::<T>::get(domain_lc.clone(), key.clone())
+            {
+                (value_with_deposit_info.1, value_with_deposit_info.2)
+            } else {
+                (owner.clone(), 0u32.into())
+            };
 
-            if let Some(new_value) = &new_value_opt {
-                DomainByInnerValue::<T>::insert(&meta.owner, new_value, domain_lc.clone());
-            }
+            let new_deposit: BalanceOf<T> = if reserve_deposit {
+                let new_deposit = Self::calc_record_deposit(key.clone(), value_opt.clone());
 
-            RegisteredDomains::<T>::mutate(&domain_lc, |meta_opt| {
-                if let Some(stored_meta) = meta_opt {
-                    stored_meta.inner_value = new_value_opt;
+                if owner == old_depositor {
+                    Self::try_reserve_deposit(&owner, old_deposit, new_deposit)?;
+                } else {
+                    <T as Config>::Currency::reserve(&owner, new_deposit)?;
+                    let err_amount = <T as Config>::Currency::unreserve(&old_depositor, old_deposit);
+                    debug_assert!(err_amount.is_zero());
                 }
+
+                new_deposit
+            } else {
+                old_deposit
+            };
+
+            DomainRecords::<T>::mutate_exists(domain_lc.clone(), key.clone(), |current_opt| {
+                *current_opt = value_opt.clone().map(|value| (value, owner.clone(), new_deposit));
             });
 
-            Self::deposit_event(Event::DomainMetaUpdated { who: meta.owner, domain: domain_lc });
+            Self::deposit_event(Event::DomainRecordUpdated {
+                account: owner,
+                domain: domain_lc,
+                key,
+                value: value_opt,
+                deposit: new_deposit,
+            });
             Ok(())
         }
 
+        pub(crate) fn calc_record_deposit(
+            key: RecordKey<T>,
+            value_opt: Option<RecordValue<T>>,
+        ) -> BalanceOf<T> {
+            let num_of_bytes: u32 =
+                if let Some(value) = value_opt {
+                    key.len().saturating_add(value.len()) as u32
+                } else {
+                    return 0u32.into();
+                };
+
+
+            T::RecordByteDeposit::get().saturating_mul(num_of_bytes.into())
```

Comment:
Should we multiply `RecordByteDeposit` by `num_of_bytes`, or should we add `num_of_bytes` to `RecordByteDeposit`? What is the standard in Substrate / Polkadot code?

Reply:
Why add, if we have 10 bytes and the byte worth is 3 SUB. Then the total deposit is 30 SUB. 10 * 3 = 30